smslibrary/src/main/java/com/eis/communication/Peer.java

```diff
@@ -1,25 +1,58 @@
 package com.eis.communication;
 
-import androidx.annotation.NonNull;
-
-import java.io.Serializable;
+import com.eis.communication.exception.InvalidPeerException;
 
 /**
- * Represents a user in the network
- * @param <A> The type of address used by the network
+ * @author Mattia Fanan
+ * abstarcton of peer class
+ * @param T type of address
  */
-public interface Peer<A> extends Serializable {
+public abstract class Peer<T> {
```

Comment:
Can't we have a full set of interfaces and then have something like `AbstractPeer<T>`? It's clearer.

Reply:
I’m not fully convinced
Everyone i found in the net says that interfaces and abstract classes are more or less the same
You use abstract for implementing some default behavior 
And interfaces if you don’t know the default behavior a priori